File: javaprojects/client/src/main/java/minigames/client/ClientRegistry.java
package minigames.client;


import java.util.HashMap;

/**
 * Holds information on which games are available and can be served up to clients.
 */
public class ClientRegistry {

    private HashMap<String, GameClient> gameClients = new HashMap<>();

    /**
     * Called by your GameServer to register it as being available to play
     * @param name 
     * @param gs
     */
    public void registerGameClient(String name, GameClient gc) {
        gameClients.put(name, gc);
    }

    public GameClient getGameClient(String name) {
        return gameClients.get(name);
    }

}

File: scalaprojects/scalafx-client/src/main/scala/minigames/scalafxclient/ClientRegistry.scala
package minigames.scalafxclient

/**
 * Holds information on which games are available and can be served up to clients.
 */
def getGameClient(name:String):Option[GameClient] = name match {
    case "MuddleText" => Some(minigames.muddletext.MuddleText())
    case _ => None

}